project_sla/analytic_account.py

```diff
@@ -44,15 +44,16 @@ def _reapply_sla(self, cr, uid, ids, recalc_closed=False, context=None):
             ctrl_models = set([sla.control_model for sla in contract.sla_ids])
             for model_name in ctrl_models:
                 model = self.pool[model_name]
-                domain = [] if recalc_closed else [('stage_id.fold', '=', 1)]
+                base = [] if recalc_closed else [('stage_id.fold', '=', 0)]
+                doc_ids = []
                 if 'analytic_account_id' in model._columns:
```

Comment:
Here may be added something like:

```python
if 'analytic_account_id' in model._columns and 'project_id' in model._columns:
    domain.append('|')
```

This will add 'or' logic if there are both columns in model.